File: evidence/generated-systems/angular/crm-on-angular/src/app/models/Team.js

// Define collection and schema for Team
export interface Team {
    name:
	type : string
    Organization:
	type : Schema.Types.ObjectId
    Users:
 	type : [{ type: Schema.Types.ObjectId, ref: 'User' }]
    Accounts:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Account' }]
    Opportunities:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Opportunity' }]
    Cases:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Case_' }]
    Campaigns:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Campaign' }]
    TeamType:
 	type : String
#
    collection: 'teams'
}

File: evidence/generated-systems/golang/crm-on-golang/crm-on-golang/internal/model/Team.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// Team Declaration
//==============================================================
type Team struct {
    gorm.Model
     Name                                    string
    OrganizationId         *uint
    Organization           *Organization `gorm:"foreignKey:OrganizationId"`
     Users           []User `gorm:"foreignKey:UsersFromTeamId"`
     Accounts           []Account `gorm:"foreignKey:AccountsFromTeamId"`
     Opportunities           []Opportunity `gorm:"foreignKey:OpportunitiesFromTeamId"`
     Cases           []Case_ `gorm:"foreignKey:CasesFromTeamId"`
     Campaigns           []Campaign `gorm:"foreignKey:CampaignsFromTeamId"`
    TeamType                      TeamType

// parent associations as their child

}

